src/main/java/seedu/address/logic/commands/DeleteSupplierCommand.java

```diff
@@ -60,24 +65,45 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         if (deleteSupplierGoodName.getGoodNames().equals(Optional.empty())) {
-            Supplier supplierToDelete = lastShownList.get(index.getZeroBased());
-            model.deleteSupplier(supplierToDelete);
+            Supplier supplierWhoHasGoodToDelete = lastShownList.get(index.getZeroBased());
+            model.deleteSupplier(supplierWhoHasGoodToDelete);
             model.commit();
-            return new CommandResult(String.format(MESSAGE_DELETE_SUPPLIER_SUCCESS, supplierToDelete));
+            return new CommandResult(String.format(MESSAGE_DELETE_SUPPLIER_SUCCESS, supplierWhoHasGoodToDelete));
 
         } else {
-            Supplier supplierToDelete = lastShownList.get(index.getZeroBased());
-            Supplier editedSupplier = supplierToDelete;
-            if (!editedSupplier.removeGood(deleteSupplierGoodName.goodNames.iterator().next())) {
-                return new CommandResult(String.format(MESSAGE_COULD_NOT_FIND_GOOD, supplierToDelete));
+            Supplier supplierWhoHasGoodToDelete = lastShownList.get(index.getZeroBased());
+            Supplier editedSupplier = createEditedSupplier(supplierWhoHasGoodToDelete,
+                    deleteSupplierGoodName.goodNames.iterator().next());
```

Comment:
I don't understand this line (line 76), what is the purpose of iterator().next()?